testing.md

```diff
@@ -3,52 +3,61 @@
 ## Developing Tests
 
 To write and run the tests, you'll need to set up your testing environment in Python.
-Be forewarned, as part of the requirements,
-you have to install the `docassemble` package, which is complex. If you have trouble downloading that
-package, take a look at the [installation instructions](https://docassemble.org/docs/installation.html)
-or post your question in the [Docassemble Slack channel](docassemble.slack.com).
 
 ### Set up a virtual environment
 
-First, set up a virtual environment called `docassemble` to hold all the packages
-related to this repository.
+First, set up a virtual environment in your command prompt called `venv` to hold all the packages related to this repository.
 This will keep your code clean, and make sure this repository does not interfere with
 your other projects.
 
+Run these lines in your command prompt.
+
 ```
-$ cd docassemble/assemblylinewizard
-$ pip3 install virtualenv
-$ virtualenv -p $(which python3) docassemble
-$ source docassemble/bin/activate
-$ pip install -r requirements.txt
+pip3 install virtualenv
+virtualenv -p $(which python3.8) venv
+source venv/bin/activate
+pip3 install --upgrade pip
+pip3 install -r docassemble/assemblylinewizard/requirements.txt
+pip3 install --editable .
 ```
 
+<!-- I had to run `virtualenv -p /usr/local/opt/python@3.8/bin/python3 venv` for the second line -->
+
 (I used `pip3` and `python3` above, because I have both Python 2 and Python 3 on my machine, but you
 can use just `pip` and `python` if you only have Python 3.)
 
 If your errors show something like `mysql_config: command not found` it means you're missing
 `mysql`, which is a dependency. You can search how to install it for your system.
 
-Then, everytime you work on this project, enter your virtual environment with
+Another common error is something like `ImportError: pycurl: libcurl link-time ssl backend (openssl) is different from compile-time ssl backend (none/other)`. https://stackoverflow.com/a/21099222/14144258 may help give you direction.
 
+If you want to stop here after setup, you can exit the virtual environment with
 ```
-$ source docassemble/bin/activate
+deactivate
 ```
 
-and everytime you finish working on this project, exit your virtual environment with
+
+### Run the tests
+
+Everytime you want to work on testing, enter your virtual environment with
 
 ```
-$ deactivate
+$ source venv/bin/activate
 ```
 
-### Run the tests
+To run the tests, make sure your command prompt is in your project's directory and use one the following commands using your own file and class names where needed. These are listed in order of "runs all tests" to "runs one test":
+
+```
+python3 -m unittest discover
+python3 -m unittest test_file
+python3 -m unittest test_file.TestClass
+python3 -m unittest test_file.TestClass.test_method
+```
```

Comment:
`unittest discover` works really well, but I don't think the other options work reliably when listing just the file. You need to list the whole module path, i.e. `docassemble.assemblylineweaver.test_file`, etc. You not only need to be in the same directory as the file for it work, but because we're importing other modules in that same directory, the tests will fail with:
```
Traceback (most recent call last):
  File "/usr/lib/python3.8/unittest/loader.py", line 154, in loadTestsFromName
    module = __import__(module_name)
  File "/home/brycew/Developer/LITLab/docassemble-assemblylinewizard/docassemble/assemblylinewizard/test_fill_in_field_attributes.py", line 2, in <module>
    from .interview_generator import DAField
ImportError: attempted relative import with no known parent package
```

Reply:
Updated, though I changed `assemblylineweaver` to `assemblylinewizard`